File: facebook-hacker-cup/2011/double-squares/double-squares.js
const fs = require('fs')

let content = fs.readFileSync('double-squares.in', 'utf8')
content = content.split('\n')
content = content.map(item => parseInt(item)).filter(item => Number.isInteger(item))

const N = content[0]
content.splice(0, 1)

for (let i = 0; i < content.length; i++) {
  let count = 0
  if (content[i] === 0) console.log(`Case #${i + 1}: 1`)
  else {
    let bound = parseInt(Math.ceil(Math.sqrt(content[i] / 2)))
    for (let a = 0; a < bound; a++) {
      let b = Math.sqrt(content[i] - (a * a))
      if (b === Math.floor(b)) count++
    }
    console.log(`Case #${i + 1}: ${count}`)
  }
}

File: facebook-hacker-cup/2011/double-squares/double-squares.py
import math

fileIn = open('double-squares.in', 'r')
fileOut = open('double-squares.out', 'w')
fileContent = fileIn.read().splitlines()
T = fileContent.pop(0)

for i, X in enumerate(fileContent):
  count = 0
  X = int(X)
  if X == 0:
    fileOut.write('Case #'+ str(i+1) + ': 1\n')
  elif X == 1:
    fileOut.write('Case #'+ str(i+1) + ': 1\n')
  elif X == 2:
    fileOut.write('Case #'+ str(i+1) + ': 1\n')
  elif X == 3:
    fileOut.write('Case #'+ str(i+1) + ': 0\n')
  else:
    bound = int(math.sqrt(X))
    for A in range(0, bound+1):
      for B in range(A, bound+1):
        if A * A + B * B > X:
          break
        if A * A + B * B == X:
          count=count+1
    fileOut.write('Case #'+ str(i+1) + ': ' + str(count) + '\n')
fileIn.close() 
fileOut.close() 
